storagetransfer/get_transfer_job_with_retries_test.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2021 Google LLC
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import warnings
+
+import backoff
+from google.api_core.exceptions import RetryError
+from google.cloud import storage_transfer
+from google.protobuf.duration_pb2 import Duration
+from googleapiclient.errors import HttpError
+import pytest
+
+import get_transfer_job_with_retries
+import get_transfer_job_with_retries_apiary
+
+project_id = os.environ["GOOGLE_CLOUD_PROJECT"]
+
+
+@pytest.fixture()
+def transfer_job():
+    # Create job
+    client = storage_transfer.StorageTransferServiceClient()
+    transfer_job = {
+        "description": "Sample job",
+        "status": "ENABLED",
+        "project_id": project_id,
+        "schedule": {
+            "schedule_start_date": {
+                "day": 1, "month": 1, "year": 2000},
+            "start_time_of_day": {
+                "hours": 0, "minutes": 0, "seconds": 0},
+        },
+        "transfer_spec": {
+            "gcs_data_source": {
+                "bucket_name": f"{project_id}-storagetransfer-source"},
```

Comment:
We are moving away from using the hard-coded storagetransfer-source and storagetransfer-sink buckets. Instead, the tests should spin up a GCS buckets, grant them the necessary permissions, and delete them at the end. They should also create and delete an AWS bucket. This will allow people to check out the code and run the tests without needing to do setup in their environments. You can see the Java PR or the quickstarts in any language for an example on how to do this